R/email.R

```diff
@@ -68,3 +64,31 @@ send_email <- function(to, email, cc = "metacheck-support@sub.uni-goettingen.de"
 xlsx_path <- function(session_id = NULL) {
   fs::path_temp(paste0(session_id, "-license_df.xlsx"))
 }
+
+#' Make Spreadsheet attachment
+#'
+#' @param .md list, returned from [cr_compliance_overview()]
+#' @param dois character, submitted dois
+#' @param session_id link spreadsheet to R session
+#'
+#' @importFrom writexl write_xlsx
+#' @export
+md_data_attachment <- function(.md = NULL, session_id = NULL, dois = NULL) {
+  if(is.null(.md))
+    stop("No Crossref Data")
+  excel_spreadsheet <- list(`Übersicht` = .md$cr_overview,
+                            `CC-Lizenzen` = .md$cc_license_check)
+  if (!is.null(.md$cr_tdm)) {
+    excel_spreadsheet[["TDM"]] <- .md$tdm
+  }
+  if (!is.null(.md$cr_funder)) {
+    excel_spreadsheet[["Förderinformationen"]] <- .md$funder_info
+  }
+  if (!length(tolower(dois) %in% tolower(.md$cr_overview$doi)) > 0)
+    excel_spreadsheet[["Nicht-indexierte DOIs"]] <- tibble::tibble(
+      mutate(missing_dois = !tolower(dois) %in% tolower(.md$cr_overview$doi))
+    )
```

Comment:
I think this might be a bit hard to check/maintain because:
- the function makes unexplicated/untested assumptions about the inner structure of `.md` (i.e. if `.md$funder_info` were renamed, it would break)
- the conditionals aren't very expressive; we don't really mean that `!is.null()` is a very special case, we just generally mean that empty sheets shouldn't be included
- the calculation of missing dois doesn't really belong in here.

I'll refactor this to show what I mean. 
Hope that makes sense.